src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.dao.DriverDao;
+import mate.exception.AuthenticationException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject private DriverDao driverDao;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverDao.findByLogin(login);
+        if (driver.isEmpty()) {
```

Comment:
single if statement is enough here if you modify check inside of it

Reply:
+